requester_test.go

```diff
@@ -195,6 +196,34 @@ func TestRequester_CustomMiddleware(t *testing.T) {
 	assert.Equal(t, 200, resp.StatusCode)
 }
 
+func TestRequester_DoNotReplaceTransport(t *testing.T) {
+	remoteTS := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {
+		t.FailNow()
+	}))
+	t.Cleanup(remoteTS.Close)
+
+	caughtReq := false
+	ts := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {
+		caughtReq = true
+	}))
+	t.Cleanup(ts.Close)
+	tsURL, err := url.Parse(ts.URL)
+	require.NoError(t, err)
+
+	redirectingRoundTripper := middleware.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		r.URL = tsURL
+		return http.DefaultTransport.RoundTrip(r)
+	})
+
+	rq := New(http.Client{Transport: redirectingRoundTripper}).Client()
+	req, err := http.NewRequest("GET", remoteTS.URL, nil)
+	require.NoError(t, err)
+	resp, err := rq.Do(req)
+	require.NoError(t, err)
+	assert.Equal(t, 200, resp.StatusCode)
+	assert.True(t, caughtReq)
```

Comment:
I think, this direct access to caughtReq may cause some linters to complain and my evne fail with -race flag in some cases as it set in ts hadnler's goroutine and read directly wihout any sync. I's suggest wrap it with atomic